packages/twenty-server/src/engine/api/graphql/graphql-query-runner/parsers/graphql-selected-fields/graphql-selected-fields.parser.ts

```diff
@@ -0,0 +1,127 @@
+import { FieldMetadataInterface } from 'src/engine/metadata-modules/field-metadata/interfaces/field-metadata.interface';
+
+import { GraphqlSelectedFieldsRelationParser } from 'src/engine/api/graphql/graphql-query-runner/parsers/graphql-selected-fields/graphql-selected-fields-relation.parser';
+import { ObjectMetadataMap } from 'src/engine/api/graphql/graphql-query-runner/utils/convert-object-metadata-to-map.util';
+import { compositeTypeDefinitions } from 'src/engine/metadata-modules/field-metadata/composite-types';
+import { isCompositeFieldMetadataType } from 'src/engine/metadata-modules/field-metadata/utils/is-composite-field-metadata-type.util';
+import { CompositeFieldMetadataType } from 'src/engine/metadata-modules/workspace-migration/factories/composite-column-action.factory';
+import { isRelationFieldMetadataType } from 'src/engine/utils/is-relation-field-metadata-type.util';
+import { capitalize } from 'src/utils/capitalize';
+
+export class GraphQLSelectedFieldsParser {
+  private graphqlSelectedFieldsRelationParser: GraphqlSelectedFieldsRelationParser;
+
+  constructor(objectMetadataMap: ObjectMetadataMap) {
+    this.graphqlSelectedFieldsRelationParser =
+      new GraphqlSelectedFieldsRelationParser(objectMetadataMap);
+  }
+
+  /**
+   * Parses the selected GraphQL fields and returns the fields to select and the relations to fetch.
+   *
+   * @param graphqlSelectedFields - The selected GraphQL fields.
+   * @param fieldMetadataMap - A map of field metadata.
+   * @returns An object containing the fields to select and the relations to fetch.
+   */
+  parse(
+    graphqlSelectedFields: Partial<Record<string, any>>,
+    fieldMetadataMap: Record<string, FieldMetadataInterface>,
+  ): { select: Record<string, any>; relations: Record<string, any> } {
+    const result: {
+      select: Record<string, any>;
+      relations: Record<string, any>;
+    } = {
+      select: {},
+      relations: {},
+    };
+
+    for (const [fieldKey, fieldValue] of Object.entries(
+      graphqlSelectedFields,
+    )) {
+      if (this.isConnectionField(fieldKey, fieldValue)) {
+        const subResult = this.parse(fieldValue, fieldMetadataMap);
+
+        Object.assign(result.select, subResult.select);
+        Object.assign(result.relations, subResult.relations);
+        continue;
+      }
+
+      const fieldMetadata = fieldMetadataMap[fieldKey];
+
+      if (!fieldMetadata) continue;
```

Comment:
Mmm, technically here I can't throw because graphqlSelectedFields can contain things that are not field metadata and we simply want to skip them and not parse/convert such as __typename, cursor, etc...